### PR TITLE
Logout errors with multiple providers

### PR DESCRIPTION
**Pull request by ebarriosjr**
## Hey, I just made a Pull Request!

I was having the problem that when you use multiple logins (in my case google and github) on Sign out the githubSession key on the local storage does not get deleted. This causes that on the next logging the auth system gets confused. I thought about clearing the entire local storage but this will remove user preferences for the page too (like the dark mode selection, pinned bar, etc)

Maybe this is not the right way to do it. But it does work :)

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Review by Rugvip — CHANGES_REQUESTED** [review]
Yep, not the way to do it x)

Seems like `AuthSessionStore` should be removing this key, which in turn should be called via the `signOut` method in the `SignInPage` provider.

Do you have some steps to reproduce the issue?

**Comment by ebarriosjr** [comment]
I tried @Rugvip  jejejeje... I can tell you exactly how it happens:
1. Log in with google from the main login page.
2. Log in with github via the side navbar.
3. Sign out via the button on the side navbar.
4. Log back in with google from the main login page.
5. If you check the localstorage you can still see the gitbubSession key there, but the login indicator in the nav bar is not active.
6. If you try to use any plugin that uses github (it may work depending if your token expired or not)

**Comment by ebarriosjr** [comment]
@Rugvip I added the refresh token to the github app. Could you take a look and give me your opinion?

File: packages/core-api/src/apis/implementations/auth/github/GithubAuth.ts [hunk 1]
```diff
@@ -87,22 +91,25 @@ class GithubAuth implements OAuthApi, SessionApi {
       },
     });
 
-    const sessionManager = new StaticAuthSessionManager({
+    const sessionManager = new RefreshingAuthSessionManager({
```
Comment (Rugvip) [inline packages/core-api/src/apis/implementations/auth/github/GithubAuth.ts:94]: Not sure about removing the `StaticAuthSessionManager` tbh, I'm thinking we either want to make this client-side implementation support both GitHub Apps and OAuth Apps, or to have a client for each.

I think a first attempt could be whether we can `GithubAuth` instance can have both a `refreshingSessionManager` and a `staticSessionManager`, and then switch which one is being used automatically. It's likely that doesn't work though, and then I'd prolly try making it either an option to the static factory method, or two different static factory methods depending on the use-case.

If we do end up with having to configure the client in the app, then I think we should stick to OAuth by default, and then switch over to GitHub App by default once we build out support for that in the project.

File: packages/core-api/src/apis/implementations/auth/github/GithubAuth.ts [hunk 2]
```diff
@@ -67,6 +70,7 @@ class GithubAuth implements OAuthApi, SessionApi {
     environment = 'development',
     provider = DEFAULT_PROVIDER,
     oauthRequestApi,
+    defaultScopes = ['read:user'],
```
Comment (Rugvip) [inline packages/core-api/src/apis/implementations/auth/github/GithubAuth.ts:73]: Don't think this is needed?

File: packages/core-api/src/apis/implementations/auth/github/GithubAuth.ts [hunk 3]
```diff
@@ -87,22 +91,25 @@ class GithubAuth implements OAuthApi, SessionApi {
...
 
-    return new GithubAuth(authSessionStore);
+    return new GithubAuth({ sessionManager });
   }
 
-  constructor(private readonly sessionManager: SessionManager<GithubSession>) {}
+  private readonly sessionManager: SessionManager<GithubSession>;
+
+  constructor(options: Options) {
+    this.sessionManager = options.sessionManager;
```
Comment (Rugvip) [inline packages/core-api/src/apis/implementations/auth/github/GithubAuth.ts:111]: Keep this as it was

**Review by Rugvip — COMMENTED** [review]
👍 , Think backend should be good, but comments on the frontend bit

File: plugins/auth-backend/src/providers/github/provider.ts [hunk 4]
```diff
@@ -126,20 +141,84 @@ export class GithubAuthProvider implements OAuthHandlers {
...
+    const { profile } = response;
+
+    if (!profile.email) {
+      throw new Error('Github profile contained no email');
+    }
+
+    try {
+      const user = await this.identityClient.findUser({
+        annotations: {
+          'github.com/email': profile.email,
```
[the remark this answers is not in the document]
Comment (freben) [reply plugins/auth-backend/src/providers/github/provider.ts:194]: specifically this https://github.com/spotify/backstage/blob/master/docs/features/software-catalog/well-known-annotations.md#githubcomuser-login

**Comment by benjdlambert** [comment]
@ebarriosjr is this still something that you could fix up for us?

**Comment by freben** [comment]
one more ping - @ebarriosjr is this something you'd be able to continue on?

**Comment by Rugvip** [comment]
Gonna close this as I believe #3770 fixed the underlying issue. Let's reopen if that's not the case

**Pull request closed without merging**